### PR TITLE
[mtouch] Enable dsym even for aot'ed images when interpreter is enabled

### PR DESCRIPTION
**Pull request by BrzVlad**
Fixes https://github.com/mono/mono/issues/13975

### CODE REVIEW

File: tools/mtouch/Application.cs [hunk 1]
```diff
@@ -514,7 +514,11 @@ public bool UseDlsym (string assembly)
 			if (EnableLLVMOnlyBitCode)
 				return false;
 
-			if (IsInterpreted (Assembly.GetIdentity (assembly)))
+			// Even if this assembly is aot'ed, if we are using the interpreter we can't yet
+			// guarantee that code in this assembly won't be executed in interpreted mode,
+			// which can happen for virtual calls between assemblies, during exception handling
+			// etc. We make sure we don't strip away symbols needed for pinvoke calls.
+			if (UseInterpreter)
```
Comment (spouliot) [inline tools/mtouch/Application.cs]: where is that limitation tracked ?
please add link so we can revert this later

Comment (lewurm) [reply tools/mtouch/Application.cs]: I don't think this limitation is fixable

Comment (BrzVlad) [reply tools/mtouch/Application.cs]: Created issue on mono side. If only the virtual calls produce this issue, it might be fixable.

Comment (spouliot) [reply tools/mtouch/Application.cs]: thanks, at least we know where to discuss (or find discussions) about this 😄

**Pull request merged**